### PR TITLE
Added support for LogicalRDDs in spark plan visitors

### PR DESCRIPTION
**Pull request by collado-mike**
Introduced a new LogicalRDDVisitor to address cases where an RDD is converted to a DataFrame. It only works with HadoopRDDs and matches input Paths with the same logic used for `HadoopFsRelation`s.

Also refactored the `JDBCRelation` and `HadoopFsRelation` visitors into a single `LogicalRelationVisitor`, with additional handling to match LogicalRelations that have a `BaseRelation` that is unknown. This allows reporting of _some_ dataset, with a description that hopefully informs more about the actual relations being used so that visitors can be added specifically. 

As a minor change, I removed references to the Hamcrest matchers and replaced with AssertJ in the modified test class.

### CODE REVIEW

File: integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/LogicalRelationVisitor.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+ * LogicalRelation}. The {@link org.apache.spark.sql.sources.BaseRelation} is tested for known
+ * types, such as {@link HadoopFsRelation} or {@link JDBCRelation}s, as those are easy to extract
+ * exact dataset information.
+ *
+ * <p>For {@link HadoopFsRelation}s, it is assumed that a single directory maps to a single {@link
+ * Dataset}. Any files referenced are replaced by their parent directory and all files in a given
+ * directory are assumed to belong to the same {@link Dataset}. Directory partitioning is currently
+ * not addressed.
+ *
+ * <p>For {@link JDBCRelation}s, {@link * Dataset} naming expects the namespace to be the JDBC
```
Comment (wslulciuc) [inline integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/LogicalRelationVisitor.java:31]: _Typo:_ Drop the `*` in `{@link * Dataset}`?

File: integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/LogicalRDDVisitor.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            rdd -> {
+              Path[] inputPaths = FileInputFormat.getInputPaths(rdd.getJobConf());
+              Configuration hadoopConf = rdd.getConf();
+              return Arrays.stream(inputPaths).map(p -> PlanUtils.getDirectoryPath(p, hadoopConf));
+            })
+        .distinct()
+        .map(
+            p -> {
+              // TODO- refactor this to return a single partitioned dataset based on static
+              // static partitions in the relation
```
Comment (wslulciuc) [inline integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/LogicalRDDVisitor.java:67]: _Typo_: Extra _static_ in comment

File: integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/LogicalRelationVisitor.java (hunk 1 above)
Comment (collado-mike) [reply integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/LogicalRelationVisitor.java]: 👍🏽

File: integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/LogicalRelationVisitor.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                  DatasetFacet.builder()
+                      .description(logRel.simpleString())
+                      .schema(PlanUtils.schemaFacet(logRel.schema()))
+                      .build())
+              .build());
+    }
+  }
+
+  private List<Dataset> handleCatalogTable(LogicalRelation logRel) {
+    CatalogTable catalogTable = logRel.catalogTable().get();
```
Comment (wslulciuc) [inline integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/LogicalRelationVisitor.java:95]: It is _safe_ to assume the catalog table would be present?

File: integrations/spark/src/test/java/marquez/spark/agent/lifecycle/SparkReadWriteIntegTest.java [hunk 4]
```diff
@@ -284,4 +287,64 @@ public void testInsertIntoDataSourceDirVisitor(@TempDir Path tempDir)
...
+    assertThat(completeEvent).hasFieldOrPropertyWithValue("eventType", "COMPLETE");
+    assertThat(completeEvent.getInputs())
+        .singleElement()
+        .hasFieldOrPropertyWithValue("name", csvPath)
+        .hasFieldOrPropertyWithValue("namespace", "file");
+
+    assertThat(completeEvent.getOutputs())
+        .singleElement()
+        .hasFieldOrPropertyWithValue("name", outputPath)
+        .hasFieldOrPropertyWithValue("namespace", "file");
```
Comment (wslulciuc) [inline integrations/spark/src/test/java/marquez/spark/agent/lifecycle/SparkReadWriteIntegTest.java:348]: _Minor:_ The static Mockito calls can be imported statically

File: integrations/spark/META-INF/MANIFEST.MF [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Manifest-Version: 1.0
+Premain-Class: marquez.spark.agent.MarquezAgent
+Agent-Class: marquez.spark.agent.MarquezAgent
+Created-By: Gradle 6.8.3
+Built-By: collado
+Build-Jdk: 15.0.1
+Implementation-Title: spark
+Implementation-Version: 0.13.2-SNAPSHOT
```
Comment (wslulciuc) [inline integrations/spark/META-INF/MANIFEST.MF:8]: Should this `MANIFEST.MF` be part of the PR?

**Comment by wslulciuc** [comment]
I think the file `integrations/airflow/tests/test_dags/test_bash_dag.py` was accidentally checked in?

**Review by wslulciuc — APPROVED** [review]
Really great work here, @collado-mike with solid tests ❤️

File: integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/LogicalRelationVisitor.java (hunk 3 above)
Comment (collado-mike) [reply integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/LogicalRelationVisitor.java:95]: We check that it is defined on line 73

File: integrations/spark/META-INF/MANIFEST.MF (hunk 5 above)
Comment (collado-mike) [reply integrations/spark/META-INF/MANIFEST.MF:8]: Nope :)

**Pull request merged**